### ISSUE TITLE
Simple combined write & read example?

### ISSUE THREAD

**Issue by brainstorm** [marcelbuesing/socketcan-isotp#2]
Hello again @marcelbuesing, thanks again for the fix on PR #1 , this helps me keep going with your crate :)

I hope this problem will be easy to solve, but I cannot figure it out. I have written the following snippet with your crate and the `embedded-can` branch:

```rust
use socketcan_isotp::{self, IsoTpSocket, ExtendedId};

fn main() -> Result<(), socketcan_isotp::Error> {
    // OEM CAN IDs
    let rx_canid = ExtendedId::new(0xCAFECAFE).expect("Invalid src CAN ID");
    let tx_canid = ExtendedId::new(0xBEEFBEEF).expect("Invalid dst CAN ID");

    let mut tp_socket = IsoTpSocket::open("can0", tx_canid,
						                                   rx_canid)?;
    tp_socket.set_nonblocking(true)?;

    tp_socket.write(&[0x02, 0x11, 0x01])?; // Soft ECU reset
    tp_socket.write(&[0x02, 0x10, 0x01])?; // Establish UDS session
    tp_socket.write(&[0x02, 0x00, 0x03])?; // Security sub-function
    tp_socket.write(&[0x02, 0x27, 0x01])?; // Ask for SEED

    println!("Sent all UDS frames, trying to read response");
    let buffer = tp_socket.read()?;

    println!("read {} bytes", buffer.len());

    for x in buffer {
        print!("{:X?} ", x);
    }

    Ok(())
}
```

With `candump` in another terminal, I can see that the frames are sent successfully but I cannot see the responses?

In other words, the `tp_socket.read()?;` seems to block forever then I looked at the API and noticed the `tp_socket.set_nonblocking(true)?;` which I then added above in the snippet, but then I get:

```
Sent all UDS frames, trying to read response                                    
Error: Io { source: Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" } }                                                            
```

Using `cansend` and `candump` from `can-utils` I can have both reading and writing working, so at least I know this should be possible somehow within the same program... `cansend` itself does not read responses after sending frames, which would be handy, but I digress.

What am I doing wrong here with the socketcan-isotp snippet shown above? Do I need to enable some flag to be able to do this? Or perhaps use two (can0/can1) interfaces for reading and writing separatedly instead (sounds overkill)?

**Comment by brainstorm** [comment]
After doing some searching, it seems that the example above needs to involve `SIOCGSTAMP` flag and timestamp management in order to work properly?:

https://stackoverflow.com/questions/67190849/socketcan-read-data-after-write
https://stackoverflow.com/questions/21135392/socketcan-continuous-reading-and-writing/55758361

I wonder if socketcan-rs or this crate should have an abstraction for this to work more transparently (such as the intuitive `.set_nonblocking()` method mentioned above).

**Comment by marcelbuesing** [comment]
I just added an uds example to the examples folder. You can test it using this [uds example server](https://github.com/zombieCraig/uds-server) implementation. Btw not sure if you know `isotpdump`i think i feel it's quite helpful in addition to candump. E.g. here: `isotpdump -u -s 7E8 vcan0 -d 77A -c`.

I'm sure there is a better way without having to resort to using a separate reader thread and multiple socket instances, I don't know how though, if you figure it out I'd be glad to hear it though.

**Comment by brainstorm** [comment]
Thank you so much @marcelbuesing, you are the best! :D

Not only it worked like a charm again, but I wasn't aware of the isotpdump `-u` flag because the latest RaspberrypyOS still ships with a `can-utils` from 2018, so I recompiled can-utils from its `master` branch and noticed that flag, I use it now, sooo handy, thanks!

I had some timing issues with the real hardware because your crate was sending packets too fast (good thing!), so I introduced some minor pauses between .write(s) on the main thread (the quick and dirty `candump` and `isotpsend` version was slow enough to not have that problem), from comments on my code:

```
    // The sending is apparently too fast with Rust and the UDS state machine does not transition properly on real hardware
    // The following numbers are deltas between the BASH script spawning candump/isotpsend commands and non-thread-sleep Rust version with socketcan-isotp:
    /*
	35006  <--- >30000ms between UDS messages
	35674
	35032

	4568	<--- Way too fast!
	6924
	353
    */
```

On a future iteration I guess I'll transition to one of your async crates for ISOTP so that I don't have to add any crude/ugly sleep()s in the code... I'm now an official fan of your crates, keep up all this good work, kudos! ;)
